pandas/tests/series/test_period.py

```diff
@@ -50,12 +51,13 @@ def test_fillna(self):
         exp = Series([pd.Period('2011-01', freq='M'),
                       pd.Period('2012-01', freq='M')])
         tm.assert_series_equal(res, exp)
-        assert res.dtype == 'object'
+        assert res.dtype == 'Period[M]'
 
-        res = s.fillna('XXX')
-        exp = Series([pd.Period('2011-01', freq='M'), 'XXX'])
-        tm.assert_series_equal(res, exp)
-        assert res.dtype == 'object'
+        # We don't support upcasting to object on fillna.
```

Comment:
This is a breaking change. I can probably live with it though.

Reply:
Can this have a big TODO or FIXME next to it?  pandas is generally very good about not having commented-out code.